samples/ChangefeedMigrationSample/Program.cs

```diff
@@ -0,0 +1,222 @@
+                destCollection,
+                new RequestOptions { OfferThroughput = 500 });
+
+            Console.WriteLine("Change feed: total {0} doc(s)", Interlocked.Add(ref s_totalDocs, docs.Count));
+            foreach (Document doc in docs)
+            {
+                Console.WriteLine(doc.ToString());
+                newClient.UpsertDocumentAsync(
+                    UriFactory.CreateDocumentCollectionUri(destCollectionLocation.DatabaseName, destCollectionLocation.CollectionName),
+                    doc);
+            }
+            return Task.CompletedTask;
+        }
+    }
+
+    class Program
+    {
+        // Modify EndPointUrl and PrimaryKey to connect to your own subscription
+        private string uri = "https://URI";
+        private string secretKey = "primaryKey";
+
+        private string dbName = "yourdbName";
+        private string monitoredCollectionName = "monitoredCollectionName";
+        private string leaseCollectionName = "leaseCollectionName";
+
+        static void Main(string[] args)
+        {
+            Console.WriteLine("ChangeFeed App");
+            Program newApp = new ChangefeedMigrationSample.Program();
+            // Thread 1 comment out for thread 2 
+            newApp.RunChangeFeedProcessor();
+            // Thread 2 comment out for thread 1 
+            // UpdateDb(EndPointUrl, PrimaryKey); 
+            Console.ReadKey();
+        }
+
+        async void RunChangeFeedProcessor()
+        {
+            // connect client 
+            DocumentClient client = new DocumentClient(new Uri(uri), this.secretKey);
+            await client.CreateDatabaseIfNotExistsAsync(new Database { Id = this.dbName });
+
+            // create monitor collection if it does not exist 
+            // WARNING: CreateDocumentCollectionIfNotExistsAsync will create a new 
+            // with reserved through pul which has pricing implications. For details
+            // visit: https://azure.microsoft.com/en-us/pricing/details/cosmos-db/
+            await client.CreateDocumentCollectionIfNotExistsAsync(
+                UriFactory.CreateDatabaseUri(this.dbName),
+                new DocumentCollection { Id = this.monitoredCollectionName },
+                new RequestOptions { OfferThroughput = 400 });
+
+            // create lease collect if it does not exist
+            // WARNING: CreateDocumentCollectionIfNotExistsAsync will create a new 
+            // with reserved through pul which has pricing implications. For details
+            // visit: https://azure.microsoft.com/en-us/pricing/details/cosmos-db/
+            await client.CreateDocumentCollectionIfNotExistsAsync(
+            UriFactory.CreateDatabaseUri(this.dbName),
+            new DocumentCollection { Id = this.leaseCollectionName },
+            new RequestOptions { OfferThroughput = 400 });
+            await StartChangeFeedHost();
+        }
+
+        async Task StartChangeFeedHost()
```

Comment:
"Start" is a bad name because it doesn't only start, it also waits and tears down.